concepts/websockets.textile

```diff
@@ -0,0 +1,591 @@
+
+As the potential for really rich experiences started to become apparent, developers naturally gravitated toward the ideal of client-server applications implemented directly in the browser. Prior to this, the standard paradigm for anything non-trivial was to build a dedicated software application, package it up with an installer, have the user download the installer and then install it natively on their machine. Needless to say, this was quite a barrier to entry for all but the tech-savvy user. Keeping the application up to date with fixes and enhancements was a challenge all of its own. It is easy to understand, then, how alluring it was to be able to build an application that neither required installation before it could be accessed, nor user training and nagging in order to iterate on the software's implementation.
+
+h3(#hatching-the-real-time-web). Hatching the real-time web
+
+bq. “Innovation is taking two things that already exist and putting them together in a new way.”
+<em>- "Tom Freston":https://en.wikipedia.org/wiki/Tom_Freston</em>
+
+
+
+When something appears to be technically possible, and the potential reward is worth the effort, we'll usually go to great lengths to bend what is available into a shape that serves our needs. Thus, developers took XMLHttpRequest and abused it to start emulating real-time back-and-forth communication between the web page and the server. Some of the techniques to do this became commonplace - "standard", even - and eventually started being referred to with the umbrella term <em>""Comet":https://en.wikipedia.org/wiki/Comet_(programming)"</em>.
+
+Probably the most common of these techniques was (is) "_long polling_":https://en.wikipedia.org/wiki/Push_technology#Long_polling. This involves opening an XMLHttpRequest connection to the server and leaving it open until ongoing communication is no longer required. Under normal circumstances when making an HTTP request, the server's response is streamed back to a client via the connection on which the request was made, the intent being to allow a browser to start rendering an HTML page while waiting for the next part of the document to be delivered by the server. With long polling, the effect of leaving an HTTP connection open is that the server can continue to deliver response data for as long as the connection remains open, and there is no technical requirement that the data be in one format or another, or that the request be closed after sending data to the client. The same applies to the HTTP request payload sent by the client. A server may begin delivering its response before the client's request data has arrived in its entirety, and the client is not strictly required to stop sending request data until it chooses to do so. This means that, just as the server may continue delivering response data for the life of the connection, the client may do the same, resulting in a defacto two-way communication stream between server and client.
```

Comment:
I don't think 'leaving it open until ongoing communication is no longer required' is factually correct. With Long Polling, the server keeps the connection open only until the data requested becomes available, after which, the response is sent but the client immediately follows up with a new request. The sentence you wrote is quite misleading. Please review this paragraph.